src/Servers/HttpSys/src/RequestProcessing/Response.cs

```diff
@@ -277,31 +276,29 @@ private void EnsureResponseStream()
     // What would we loose by bypassing HttpSendHttpResponse?
     //
     // TODO: Consider using the HTTP_SEND_RESPONSE_FLAG_BUFFER_DATA flag for most/all responses rather than just Opaque.
-    internal unsafe uint SendHeaders(HttpApiTypes.HTTP_DATA_CHUNK[]? dataChunks,
+    internal unsafe uint SendHeaders(ref UnmanagedBufferAllocator allocator,
+        Span<HttpApiTypes.HTTP_DATA_CHUNK> dataChunks,
```

Comment:
Did you also meant to allocate `HTTP_DATA_CHUNK`s in `BuildDataChunks` using `UnmanagedBufferAllocator`? 

(This change from array to span does not appear to be needed.)

Reply:
Oh right. This was another option was exploring. It didn't work out since I would have needed to update several call sites and the changes weren't really all that beneficial.

Reply:
It did not look too bad to me. I will leave it to @davidfowl to comment on how important it is to avoid managed allocations on this path. `HTTP_DATA_CHUNK[]` and `List<GCHandle>` can be switched to `UnmanagedBufferAllocator` and `Span<HTTP_DATA_CHUNK>` and `Span<GCHandle>`.

Reply:
We have similar code in IIS (https://github.com/dotnet/aspnetcore/blob/e35f2ea7994a3b18fc00c5f65c315739e41147be/src/Servers/IIS/IIS/src/Core/IO/AsyncWriteOperationBase.cs#L38-L53). I like the idea of switching this to unmanaged allocations too 👍🏾